internal/pkg/earpointer/taint.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package earpointer
+
+import (
+	"go/types"
+
+	"github.com/google/go-flow-levee/internal/pkg/source"
+	"golang.org/x/tools/go/ssa"
+)
+
+// A data structure to assist the traversal of an EAR heap.
+type heapTraversal struct {
+	partitions *Partitions
+	callees    map[*ssa.Function]bool // the functions containing the references of interest
+	visited    ReferenceSet           // the visited references during the traversal
+}
+
+// For a function, transitively get the functions called within this function.
+// Argument "depth" controls the depth of the call chain.
+// For example, return {g1,g2,g3} for "func f(){ g1(); g2() }, func g1(){ g3() }".
+func getCalleeFunctions(fn *ssa.Function, result map[*ssa.Function]bool, depth uint) {
+	if depth <= 0 {
+		return
+	}
+	for _, b := range fn.Blocks {
+		for _, instr := range b.Instrs {
+			if call, ok := instr.(*ssa.Call); ok {
+				// TODO: the call graph can be CHA, RTA, VTA, etc.
```

Comment:
Addressing this will change the public signature. (I think this is fine but @mlevesquedion should make the call.)

Reply:
This looks fine to me also.

Consider creating an issue an referencing it in the TODO, though. (I think there are other similar TODOs lying around, if you want to create the issue please also update those comments).